pynapple/core/time_series.py

```diff
@@ -915,70 +930,119 @@ def __init__(
 
         self.columns = pd.Index(c)
         self.nap_class = self.__class__.__name__
+        _MetadataBase.__init__(self, **kwargs)
         self._initialized = True
 
     @property
     def loc(self):
         return _TsdFrameSliceHelper(self)
 
     def __repr__(self):
```

Comment:
this method for repr it is quite annoying, maybe it is worth moving it into a script, which contains the funcitons for computing the  reprs of all the objects. maybe there is some shared code that help compressing